src/store/reducer.js

```diff
@@ -0,0 +1,26 @@
+const initialState = {
+  counter: 0,
+};
+
+function rootReducer(state = initialState, action) {
+  let counter;
+  switch (action.type) {
+    case 'INCREASE_COUNTER':
+      counter = state.counter + 1;
```

Comment:
don`t redeclarate a variable, your state is an object, you always should return ...state, counter: value

Reply:
all logic should be in action.